homeassistant/helpers/dispatcher.py

```diff
@@ -87,21 +115,14 @@ def async_dispatcher_send(hass: HomeAssistant, signal: str, *args: Any) -> None:
 
     This method must be run in the event loop.
     """
-    target_list: dict[
-        Callable[..., Any], HassJob[..., None | Coroutine[Any, Any, None]] | None
-    ] = hass.data.get(DATA_DISPATCHER, {}).get(signal, {})
+    if (maybe_dispatchers := hass.data.get(DATA_DISPATCHER)) is None:
+        return
+    dispatchers: _DispatcherDataType = maybe_dispatchers
+    if (target_list := dispatchers.get(signal)) is None:
+        return
 
-    run: list[HassJob[..., None | Coroutine[Any, Any, None]]] = []
-    for target, job in target_list.items():
+    for target, job in list(target_list.items()):
```

Comment:
Better to just make a copy of the items instead of appending it all to a list one at a time since the python function overhead of calling `.append()` over and over it much worse